lib/resty/session/utils.lua

```diff
@@ -1004,6 +1004,89 @@ local function has_flag(options, flag)
 end
 
 
+---
+-- Function to determine if the storage should be cleaned up.
+--
+-- @function utils.should_cleanup
+-- @treturn  boolean  true if storage should be cleaned up, false otherwise
+-- Cleanup probability varies between 0.01 (1/100) and inf, based on last
+-- cleanup timestamp
```

Comment:
This function decides whether it is time to clean up file, mysql and postgres storages. It considers the previous cleanup timestamp and randomizes the chances, with a reference time of 120 seconds.